aesm-client/src/imp/aesm_protobuf/mod.rs

```diff
@@ -107,4 +109,150 @@ impl AesmClient {
         Ok(QuoteResult::new(quote, qe_report))
     }
 
+
+    ///
+    /// Purpose of following functions is to stay as close as possible to protobuf interface.
+    /// for more usable API see AesmClient in rust-sgx/aesm-client/src/lib.rs
+    /// 
+    
+    ///
+    /// Returns number of keys stored in AESM service. This is used internally in get_supported_att_key_ids.
+    ///
+    pub fn get_supported_att_key_id_num(&self) -> Result<u32> {
+        #[allow(unused_mut)]
+        let mut req = Request_GetSupportedAttKeyIDNumRequest::new();
+        // FIXME: remove conditional compilation after resolving https://github.com/fortanix/rust-sgx/issues/31
+        #[cfg(not(target_env = "sgx"))]
+        req.set_timeout(REMOTE_AESM_TIMEOUT_US);
+        let res = self.transact(req)?;
+
+        Ok(res.get_att_key_id_num())
+    }
+
+    ///
+    /// Returns all keys supported by AESM service.
+    /// Similar functionality to sgx_get_supported_att_key_ids in page 177 at https://download.01.org/intel-sgx/sgx-linux/2.9.1/docs/Intel_SGX_Developer_Reference_Linux_2.9.1_Open_Source.pdf
+    ///
+    /// @returns structure containing set of keys that can serve as input for init_quote_ex/get_quote_size_ex operations. (see AesmKeyIds::select_algorithm_id to extract a specific key)
```

Comment:
Please don't use javadoc syntax in Rust documentation

Reply:
fixed, thanks